src/main/java/org/opensearch/flowframework/transport/ProvisionWorkflowTransportAction.java

```diff
@@ -97,7 +111,26 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
                 // Parse template from document source
                 Template template = Template.parse(response.getSourceAsString());
 
-                // TODO : Update state index entry to PROVISIONING, given workflowId
+                flowFrameworkIndicesHandler.updateWorkflowState(
+                    workflowId,
+                    ImmutableMap.of(
+                        STATE_FIELD,
+                        State.PROVISIONING,
+                        PROVISIONING_PROGRESS_FIELD,
+                        ProvisioningProgress.IN_PROGRESS,
+                        PROVISION_START_TIME_FIELD,
+                        Instant.now().toEpochMilli(),
+                        USER_OUTPUTS_FIELD,
+                        Map.of("key1", "key2")
+                    ),
+                    ActionListener.wrap(updateResponse -> {
+                        logger.info("updated workflow {} state to PROVISIONING", request.getWorkflowId());
+                        listener.onResponse(new WorkflowResponse(request.getWorkflowId()));
+                    }, exception -> {
+                        logger.error("Failed to update workflow state : {}", exception.getMessage());
```

Comment:
Same question as above.

Reply:
Then doc would still be saying "NOT_STARTED" and not have a start time and etc.

Reply:
Let's discuss in #122